### PR TITLE
Bump nix, rand, and more

### PR DESCRIPTION
**Pull request by PsiACE**
Signed-off-by: Chojan Shang <psiace@outlook.com>

### CODE REVIEW

File: src/file_pipe_log/log_file.rs [hunk 1]
```diff
@@ -30,19 +30,14 @@ const FILE_ALLOCATE_SIZE: usize = 2 * 1024 * 1024;
...
 fn from_nix_error(e: nix::Error, custom: &'static str) -> std::io::Error {
-    match e {
-        nix::Error::Sys(no) => {
-            let kind = std::io::Error::from(no).kind();
-            std::io::Error::new(kind, custom)
-        }
-        e => std::io::Error::new(std::io::ErrorKind::Other, format!("{}: {:?}", custom, e)),
-    }
+    let kind = std::io::Error::from(e).kind();
+    std::io::Error::new(kind, custom)
```
Comment (tabokie) [inline src/file_pipe_log/log_file.rs:34]: Handle the case where kind == Other?

**Review by tabokie — APPROVED** [review]
Thanks! Rest looks good.

File: src/file_pipe_log/log_file.rs (hunk 1 above)
Comment (PsiACE) [reply src/file_pipe_log/log_file.rs:34]: In nix 0.23, all error types have been unified to Errno, and it appears that the `Other` kind no longer needs to be handled

Comment (tabokie) [reply src/file_pipe_log/log_file.rs:34]: I see.

**Pull request merged**